src/services/upper-bar-manager/ui/displayed-bar/displayed-bar.component.tsx

```diff
@@ -83,7 +83,7 @@ export class DisplayedBar extends Component<DisplayedBarProps & PropsFromRedux,
 
   render(): ComponentChild {
     const { displayedControls, dropdownControls } = this.splitControlsIntoDisplayedAndDropdown();
-    return (
+    return displayedControls.length > 0 ? (
```

Comment:
what the gradient protector has to do with the top bar ? are they effecting each other ?

Reply:
the gradient is the background of the top-bar itself. if you will take a look at the related PR in ui (link is attached in the description), you'd see that i have chosen to solve it in css. either way, if the displayedBar is empty, we can avoid rendering it.

Reply:
Got it